features/dd-sdk-android-profiling/src/main/java/com/datadog/android/profiling/internal/perfetto/PerfettoProfiler.kt

```diff
@@ -42,13 +44,15 @@ internal class PerfettoProfiler(
 
     private val resultCallback: Consumer<ProfilingResult>
 
-    private val profilingStarted = AtomicBoolean(false)
+    // This flag represents which instance of this class is working for.
+    private val runningInstances: AtomicReference<Set<String>> = AtomicReference(emptySet())
```

Comment:
minor: this doesn't prevent concurrent access to the same instance `Set<String>`. If it is possible to have such access, we need to use `Collections.newSetFromMap(ConcurrentHashMap<String, Boolean>())`.

`AtomicReference` is used below mainly for `compareAndSet`, but if we use concurrent set we probably don't need `AtomicReference` (not quite sure here though without more precise checks).